pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/Utils.java

```diff
@@ -99,18 +100,19 @@ public static void uploadToBookeeper(Namespace dlogNamespace,
                                          String destPkgPath)
             throws IOException {
 
+        String encodedDestPkgPath = Codec.encode(destPkgPath);
```

Comment:
If we encode the whole path, won't we end up having a flat list of names? 

I think we should do as in topics where only the final part of topic is escaped

Reply:
make sense. will fix it.

Reply:
@merlimat `destPkgPath` is the function-name coming from user while [uploading function](https://github.com/apache/incubator-pulsar/blob/master/pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/FunctionsImpl.java#L620) and it stores under db-namespace. so, here it is encoding only function-name pkg only. so, it should be fine.